packages/drivers/odsp-driver/src/test/odspSummaryUploadManagerTests.spec.ts

```diff
@@ -0,0 +1,308 @@
+        assert(serializedTree.includes("\"id\":\"ackHandle/.app/default/component\""),
+            "Component blob should be deduped");
+    });
+
+    it("Should dedup correct blobs(with handle expansion)", async () => {
+        const rootBlob: api.ISummaryBlob = {
+            type: api.SummaryType.Blob,
+            content: JSON.stringify("root"),
+        };
+        const componentBlob: api.ISummaryBlob = {
+            type: api.SummaryType.Blob,
+            content: JSON.stringify("component"),
+        };
+
+        const appSummary: api.ISummaryTree = {
+            type: api.SummaryType.Tree,
+            tree: {
+                default: {
+                    type: api.SummaryType.Tree,
+                    tree: {
+                        header: {
+                            type: api.SummaryType.Tree,
+                            tree: {
+                                component: componentBlob,
+                                root: rootBlob,
+                            },
+                        },
+                    },
+                },
+            },
+        };
+
+        const blobTreeDedupCaches: IDedupCaches = {
+            blobShaToPath: new Map(),
+            pathToBlobSha: new Map(),
+            treesPathToTree: new Map(),
+        };
+        await odspSummaryUploadManager["convertSummaryToSnapshotTree"](
+            undefined,
+            cloneDeep(appSummary),
+            blobTreeDedupCaches,
+            ".app",
+            true,
+        );
+
+        // Now insert another blob with same content as component blob
+        appSummary.tree.default2 = {
+            type: api.SummaryType.Tree,
+            tree: {
+                component2: componentBlob,
+                header: {
+                    type: api.SummaryType.Blob,
+                    content: JSON.stringify("headerBlob"),
+                },
+            },
+        };
+        appSummary.tree.default = {
+            type: api.SummaryType.Handle,
+            handle: "default",
+            handleType: api.SummaryType.Tree,
+        };
+        odspSummaryUploadManager["blobTreeDedupCaches"] = blobTreeDedupCaches;
+        const { snapshotTree, blobs, reusedBlobs } = await odspSummaryUploadManager["convertSummaryToSnapshotTree"](
+            "ackHandle",
+            appSummary,
+            blobTreeDedupCaches,
+            ".app",
+            true,
+        );
+        const serializedTree = JSON.stringify(snapshotTree);
+        assert.strictEqual(reusedBlobs, 3, "2 reused blobs should be there");
+        assert.strictEqual(blobs, 1, "1 blob(.app/default2/header) is not deduped as content does not match");
+        assert(serializedTree.includes("\"id\":\"ackHandle/.app/default/header/root\""),
+            "Root blob should be deduped");
+        assert(serializedTree.includes("\"id\":\"ackHandle/.app/default/header/component\""),
+            "Component blob should be deduped");
+    });
```

Comment:
Given that we accumulate a bunch of state from summary to summary, it would be great to have some UT that does multiple iterations, including mostly no-op iterations (let's say tree that is reusing all children from previous summary), and iterations that slowly add new and old blobs.

Reply:
added test